src/rimage.c

```diff
@@ -115,19 +115,21 @@ int main(int argc, char *argv[])
 		return -EINVAL;
 	}
 
-	/* firmware version and build id */
+	/* firmware version: major.minor.micro */
 	if (image.fw_ver_string) {
-		ret = sscanf(image.fw_ver_string, "%hu.%hu",
+		ret = sscanf(image.fw_ver_string, "%hu.%hu.%hu",
 			     &image.fw_ver_major,
-			     &image.fw_ver_minor);
+			     &image.fw_ver_minor,
+			     &image.fw_ver_micro);
 
-		if (ret != 2) {
+		if (ret != 3) {
```

Comment:
It would be very useful to change the error message a bit. So when there is SOF + rimage mismatch (which _will_ happen in some places) the build failure message is enough to immediately make the difference between an old, 2-fields rimage versus a new, 3 fields rimage. No need to go and try to guess the git versions that were used.

```C
fprintf(stderr,
   "error: cannot parse firmware version major.minor.micro\n");
```

Reply:
@marc-hb Thank you! I updated the PR to revise the error message as you suggested. The help message for file version is also updated.